msal/tests/dev apps/DesktopTestApp/MainForm.cs

```diff
@@ -63,6 +66,8 @@ public MainForm()
 
             LoadSettings();
             Logger.LogCallback = LogDelegate;
+            LabResponse labResponse = LabUserHelper.GetLabResponseWithB2CLocalAccountProvider();
```

Comment:
We shouldn't make a web request to the lab for B2C everytime we open the app, because it is quite expensive. I believe that the way tests run today is that each test closes and reopens the app. So all our non-B2C tests make this call, which is not needed. 

Instead, you can initialize _b2CClientId = null, then when you click on B2CLogin go query the lab. If I press B2CLogin again, then _b2CClientId is initialized so we won't have to call the lab again. 

This is called lazy initialization, and it's perfectly ok to do it manually like I described. Otherwise, C# has a class for this called Lazy<T> https://docs.microsoft.com/en-us/dotnet/api/system.lazy-1?view=netframework-4.7.2 - we use it in a few places in MSAL.